staging/src/k8s.io/apiserver/pkg/admission/plugin/policy/validating/caching_authorizer.go

```diff
@@ -72,16 +76,31 @@ var _ user.Info = (interface {
 // Authorize returns an authorization decision by delegating to another Authorizer. If an equivalent
 // check has already been performed, a cached result is returned. Not safe for concurrent use.
 func (ca *cachingAuthorizer) Authorize(ctx context.Context, a authorizer.Attributes) (authorizer.Decision, string, error) {
-	serializableAttributes := authorizer.AttributesRecord{
-		Verb:            a.GetVerb(),
-		Namespace:       a.GetNamespace(),
-		APIGroup:        a.GetAPIGroup(),
-		APIVersion:      a.GetAPIVersion(),
-		Resource:        a.GetResource(),
-		Subresource:     a.GetSubresource(),
-		Name:            a.GetName(),
-		ResourceRequest: a.IsResourceRequest(),
-		Path:            a.GetPath(),
+	type SerializableAttributes struct {
+		authorizer.AttributesRecord
+		LabelSelector string
+	}
+
+	serializableAttributes := SerializableAttributes{
+		AttributesRecord: authorizer.AttributesRecord{
+			Verb:            a.GetVerb(),
+			Namespace:       a.GetNamespace(),
+			APIGroup:        a.GetAPIGroup(),
+			APIVersion:      a.GetAPIVersion(),
+			Resource:        a.GetResource(),
+			Subresource:     a.GetSubresource(),
+			Name:            a.GetName(),
+			ResourceRequest: a.IsResourceRequest(),
+			Path:            a.GetPath(),
+		},
+	}
+	// in the error case, we won't honor this field selector, so the cache doesn't need it.
+	if fieldSelector, err := a.ParseFieldSelector(); fieldSelector != nil {
+		serializableAttributes.FieldSelectorRequirements, serializableAttributes.FieldSelectorParsingErr = fieldSelector, err
+	}
+	if labelSelector, _ := a.ParseLabelSelector(); labelSelector != nil {
+		// the labels requirements have private elements so those don't help us serialize to a unique key
+		serializableAttributes.LabelSelector = labels.NewSelector().Add(labelSelector...).String()
```

Comment:
`labels.NewSelector().Add(labelSelector...)` reallocs and sorts ... is that necessary? can we stringify and join the requirements directly?

Reply:
> can we stringify and join the requirements directly?

There's no string method, but I can write one if that's your preference.

Reply:
really?

https://github.com/kubernetes/kubernetes/blob/6ac60160c5729ade462b041b170ec8ac0f1eb3bc/staging/src/k8s.io/apimachinery/pkg/labels/selector.go#L307-L310

Reply:
oh, you meant there's no Requirements#String() ... yeah, defining that to stringify the members and join with `,` and using it here seems nicer to me